### PR TITLE
Staff role access (admin except bot invite)

### PR DESCRIPTION
**Pull request by wraithfive**
## Summary
- Allow Staff role to manage everything admins can, except bot invite
- Keep bot invite restricted to actual admins (ADMINISTRATOR or MANAGE_SERVER)
- Add Staff-aware checks in admin service and controller
- Add tests for Staff role access, admin-only invites, and negative cases

## Testing
- ./build.sh
- Added new unit tests in AdminServiceBranchesTest and ServerControllerTest

## Notes
- Maintains admin-only bot invite; Staff cannot invite the bot
- No schema changes; logic only

### CODE REVIEW

File: src/test/java/com/discordbot/AdminServiceBranchesTest.java [hunk 1]
```diff
@@ -252,6 +252,94 @@ void isUserAdminInGuild_noToken() {
...
+        // User has no admin permissions
+        Map<String,Object> guild = new HashMap<>();
+        guild.put("id", "g1");
+        guild.put("name", "G");
+        guild.put("permissions", "0");
+        when(cache.get(eq("tok"), any())).thenReturn(List.of(guild));
+
+        // Even with Staff role, should return false
+        assertFalse(service.isUserActualAdminInGuild(auth, "g1"));
+    }
```
Comment (wraithfive) [inline src/test/java/com/discordbot/AdminServiceBranchesTest.java]: Updated: added Staff-role mocks so this test now proves Staff alone doesn’t grant actual-admin. Also added a new test covering non-numeric permissions parsing, which now returns false.

File: src/main/java/com/discordbot/web/service/AdminService.java [hunk 2]
```diff
@@ -188,14 +189,66 @@ public boolean isUserAdminInGuild(Authentication authentication, String guildId)
...
+
+        if (accessToken == null) {
+            return false;
+        }
+
+        List<Map<String, Object>> userGuilds = getUserGuildsFromDiscord(accessToken);
+
+        for (Map<String, Object> userGuild : userGuilds) {
+            if (guildId.equals(userGuild.get("id"))) {
+                Long permissions = Long.parseLong(userGuild.get("permissions").toString());
```
Comment (wraithfive) [inline src/main/java/com/discordbot/web/service/AdminService.java]: Handled both concerns here: now we null-check and safely parse permissions, treating unparseable values as no admin; also using primitive `long` for the bitmask.

Comment (wraithfive) [inline src/main/java/com/discordbot/web/service/AdminService.java]: Also updated the declaration to primitive `long permissions` to avoid boxing/null and keep the bitmask consistent.

**Review by wraithfive — COMMENTED** [review]
Addressed feedback: added Staff-role mocks to the negative actual-admin test, added a parse-guard test for non-numeric permissions, and hardened permission parsing with primitive long + null/format handling.
